sdk/keyvault/keyvault-admin/src/accessControlClient.ts

```diff
@@ -0,0 +1,459 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/// <reference lib="esnext.asynciterable" />
+
+import {
+  operationOptionsToRequestOptionsBase,
+  TokenCredential,
+  isTokenCredential,
+  signingPolicy,
+  createPipelineFromOptions
+} from "@azure/core-http";
+import { PageSettings, PagedAsyncIterableIterator } from "@azure/core-paging";
+
+import { challengeBasedAuthenticationPolicy } from "../../keyvault-common/src";
+import {
+  RoleAssignmentsCreateResponse,
+  RoleAssignmentsDeleteResponse,
+  RoleAssignmentsListForScopeOptionalParams
+} from "./generated/models";
+
+import {
+  CreateRoleAssignmentOptions,
+  KeyVaultRoleAssignment,
+  AccessControlClientOptions,
+  RoleAssignmentScope,
+  DeleteRoleAssignmentOptions,
+  ListRoleAssignmentsOptions,
+  ListRoleDefinitionsOptions,
+  KeyVaultRoleDefinition,
+  GetRoleAssignmentOptions
+} from "./accessControlModels";
+import { SDK_VERSION, LATEST_API_VERSION } from "./constants";
+import { KeyVaultClient } from "./generated/keyVaultClient";
+import { createSpan, setParentSpan } from "./tracing";
+import { mappings } from "./mappings";
+import { logger } from "./log";
+
+/**
+ * The KeyVaultAccessControlClient provides methods to manage
+ * access control and role assignments in any given Azure Key Vault instance.
+ * The client supports creating, retrieving and deleting roles.
+ */
+export class KeyVaultAccessControlClient {
+  /**
+   * The base URL to the vault
+   */
+  public readonly vaultUrl: string;
+
+  /**
+   * @internal
+   * @ignore
+   * A reference to the auto-generated KeyVault HTTP client.
+   */
+  private readonly client: KeyVaultClient;
+
+  /**
+   * Creates an instance of the KeyVaultAccessControlClient.
+   *
+   * Example usage:
+   * ```ts
+   * import { KeyVaultAccessControlClient } from "@azure/keyvault-admin";
+   * import { DefaultAzureCredential } from "@azure/identity";
+   *
+   * let vaultUrl = `https://<MY KEYVAULT HERE>.vault.azure.net`;
+   * let credentials = new DefaultAzureCredential();
+   *
+   * let client = new KeyVaultAccessControlClient(vaultUrl, credentials);
+   * ```
+   * @param {string} vaultUrl the URL of the Key Vault. It should have this shape: https://${your-key-vault-name}.vault.azure.net
```

Comment:
I thought we weren't going to specify param types in JSDoc anymore since it duplicates what TS does already? /cc @ramya-rao-a

Reply:
Sounds good to me! I don't remember this though. Should we open issues to clean up the existing clients?

Reply:
I'll remove the types of the params and the memeberof instances!